src/app/examples/todos/todos.reducer.ts

```diff
@@ -14,7 +14,9 @@ export enum TodosActionTypes {
 export class ActionTodosAdd implements Action {
   readonly type = TodosActionTypes.ADD;
 
-  constructor(readonly payload: { name: string }) {}
+  constructor(public payload: { id?: string; name: string }) {
+    this.payload.id = uuid();
```

Comment:
This will always set `id` to uuid even if user passes id when calling `new ActionTodosAdd()` so either we support passing of the `id` or we remove optional `id` parameter from constructor. Also, in general it's better to define optional parameters at the end instead of at the beginning.

Reply:
Ha, I was just typing the same 😅